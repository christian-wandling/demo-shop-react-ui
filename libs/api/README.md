# api

A comprehensive library for consuming the demo-shop api. Supports accessing a product catalog, user management,
shopping cart operations, and order processing.
