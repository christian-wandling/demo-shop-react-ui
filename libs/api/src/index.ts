export * from './lib/openapi';
export * from './lib/ApiContext';
export * from './lib/services/apiService';
export * from './lib/ApiProvider';
export * from './lib/services/apiService';
export * from './lib/middleware/authMiddleware';
