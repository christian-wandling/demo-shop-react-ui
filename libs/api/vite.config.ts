import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';

export default defineConfig(() => ({
  root: __dirname,
  cacheDir: '../../node_modules/.vite/libs/api',
  plugins: [react()],
  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },
  test: {
    watch: false,
    globals: true,
    environment: 'jsdom',
    reporters: ['default'],
    passWithNoTests: true,
    coverage: {
      reportsDirectory: './test-output/vitest/coverage',
      enabled: false,
    },
  },
}));
