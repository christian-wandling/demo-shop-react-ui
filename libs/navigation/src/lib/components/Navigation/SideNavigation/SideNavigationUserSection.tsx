import { UserNavigationProps } from '../../../models/userNavigationProps';

export const SideNavigationUserSection = ({ user, onLogin, onLogout, onRegister }: UserNavigationProps) => {
  if (user) {
    return (
      <div className="lg:hidden space-y-6 border-t border-gray-200 px-4 py-6">
        <div className="flow-root">
          <a className="-m-2 block p-2 font-medium text-xl text-gray-400">
            {user.firstname}&nbsp;{user.lastname}
          </a>
        </div>
        <div className="flow-root">
          <a
            onClick={onLogout}
            className="-m-2 block p-2 font-medium text-xl text-gray-700 hover:text-indigo-500 cursor-pointer">
            Sign out
          </a>
        </div>
      </div>
    );
  }

  return (
    <div className="lg:hidden space-y-6 border-t border-gray-200 px-4 py-6">
      <div className="flow-root">
        <a
          onClick={onLogin}
          className="-m-2 block p-2 font-medium text-xl text-gray-700 hover:text-indigo-500 cursor-pointer">
          Sign in
        </a>
      </div>
      <div className="flow-root">
        <a
          onClick={onRegister}
          className="-m-2 block p-2 font-medium text-xl text-gray-700 hover:text-indigo-500 cursor-pointer">
          Register
        </a>
      </div>
    </div>
  );
};
