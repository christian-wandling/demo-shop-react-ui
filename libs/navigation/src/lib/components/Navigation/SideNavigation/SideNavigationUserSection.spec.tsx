import { describe, expect, it, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { SideNavigationUserSection } from './SideNavigationUserSection';
import { UserResponse } from '@demo-shop-react-ui/api';

describe('SmallScreenUserNavigation', () => {
  const mockOnLogin = vi.fn();
  const mockOnLogout = vi.fn();
  const mockOnRegister = vi.fn();

  const mockUser: UserResponse = {
    email: 'email',
    id: 1,
    phone: null,
    firstname: 'John',
    lastname: 'Doe',
  };

  it('renders successfully', () => {
    const { baseElement } = render(
      <SideNavigationUserSection onLogin={mockOnLogin} onLogout={mockOnLogout} onRegister={mockOnRegister} />
    );

    expect(baseElement).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { baseElement } = render(
      <SideNavigationUserSection onLogin={mockOnLogin} onLogout={mockOnLogout} onRegister={mockOnRegister} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('renders login and register options when user is not logged in', () => {
    render(<SideNavigationUserSection onLogin={mockOnLogin} onLogout={mockOnLogout} onRegister={mockOnRegister} />);

    expect(screen.getByText('Sign in')).toBeTruthy();
    expect(screen.getByText('Register')).toBeTruthy();
    expect(screen.queryByText('Sign out')).not.toBeTruthy();
  });

  it('renders user name and logout option when user is logged in', () => {
    render(
      <SideNavigationUserSection
        user={mockUser}
        onLogin={mockOnLogin}
        onLogout={mockOnLogout}
        onRegister={mockOnRegister}
      />
    );

    expect(screen.getByText('John Doe')).toBeTruthy();
    expect(screen.getByText('Sign out')).toBeTruthy();
    expect(screen.queryByText('Sign in')).not.toBeTruthy();
    expect(screen.queryByText('Register')).not.toBeTruthy();
  });

  it('calls onLogin when Sign in is clicked', () => {
    render(<SideNavigationUserSection onLogin={mockOnLogin} onLogout={mockOnLogout} onRegister={mockOnRegister} />);

    fireEvent.click(screen.getByText('Sign in'));
    expect(mockOnLogin).toHaveBeenCalledTimes(1);
  });

  it('calls onRegister when Register is clicked', () => {
    render(<SideNavigationUserSection onLogin={mockOnLogin} onLogout={mockOnLogout} onRegister={mockOnRegister} />);

    fireEvent.click(screen.getByText('Register'));
    expect(mockOnRegister).toHaveBeenCalledTimes(1);
  });

  it('calls onLogout when Sign out is clicked', () => {
    render(
      <SideNavigationUserSection
        user={mockUser}
        onLogin={mockOnLogin}
        onLogout={mockOnLogout}
        onRegister={mockOnRegister}
      />
    );

    fireEvent.click(screen.getByText('Sign out'));
    expect(mockOnLogout).toHaveBeenCalledTimes(1);
  });

  it('renders with correct CSS classes for mobile view', () => {
    const { container } = render(
      <SideNavigationUserSection onLogin={mockOnLogin} onLogout={mockOnLogout} onRegister={mockOnRegister} />
    );

    const navContainer = container.firstChild as HTMLElement;
    expect(navContainer.classList).toContain('lg:hidden');
    expect(navContainer.classList).toContain('border-t');
    expect(navContainer.classList).toContain('border-gray-200');
  });

  it('formats the user name correctly when user is logged in', () => {
    render(
      <SideNavigationUserSection
        user={mockUser}
        onLogin={mockOnLogin}
        onLogout={mockOnLogout}
        onRegister={mockOnRegister}
      />
    );

    expect(screen.getByText('John Doe')).toBeTruthy();
  });
});
