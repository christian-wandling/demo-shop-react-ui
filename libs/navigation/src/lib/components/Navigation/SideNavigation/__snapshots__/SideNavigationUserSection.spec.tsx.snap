// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SideNavigationUserSection > matches snapshot 1`] = `
<body>
  <div>
    <div
      class="lg:hidden space-y-6 border-t border-gray-200 px-4 py-6"
    >
      <div
        class="flow-root"
      >
        <a
          class="-m-2 block p-2 font-medium text-xl text-gray-700 hover:text-indigo-500 cursor-pointer"
        >
          Sign in
        </a>
      </div>
      <div
        class="flow-root"
      >
        <a
          class="-m-2 block p-2 font-medium text-xl text-gray-700 hover:text-indigo-500 cursor-pointer"
        >
          Register
        </a>
      </div>
    </div>
  </div>
</body>
`;
